test/unit/usergroup_test.rb

```diff
@@ -29,19 +29,25 @@ class UsergroupTest < ActiveSupport::TestCase
   end
 
   def populate_usergroups
-    @u1 = User.find_or_create_by_login :login => "u1", :mail => "u1@someware.com", :firstname => "u1", :auth_source => auth_sources(:one)
-    @u2 = User.find_or_create_by_login :login => "u2", :mail => "u2@someware.com", :firstname => "u2", :auth_source => auth_sources(:one)
-    @u3 = User.find_or_create_by_login :login => "u3", :mail => "u3@someware.com", :firstname => "u3", :auth_source => auth_sources(:one)
-    @u4 = User.find_or_create_by_login :login => "u4", :mail => "u4@someware.com", :firstname => "u4", :auth_source => auth_sources(:one)
-    @u5 = User.find_or_create_by_login :login => "u5", :mail => "u5@someware.com", :firstname => "u5", :auth_source => auth_sources(:one)
-    @u6 = User.find_or_create_by_login :login => "u6", :mail => "u6@someware.com", :firstname => "u6", :auth_source => auth_sources(:one)
-
-    @ug1 = Usergroup.find_or_create_by_name :name => "ug1"
-    @ug2 = Usergroup.find_or_create_by_name :name => "ug2"
-    @ug3 = Usergroup.find_or_create_by_name :name => "ug3"
-    @ug4 = Usergroup.find_or_create_by_name :name => "ug4"
-    @ug5 = Usergroup.find_or_create_by_name :name => "ug5"
-    @ug6 = Usergroup.find_or_create_by_name :name => "ug6"
+    @u1 = User.where(:login => "u1", :mail => "u1@someware.com",
+                     :firstname => "u1", :auth_source_id => auth_sources(:one)).first_or_create
+    @u2 = User.where(:login => "u2", :mail => "u2@someware.com",
+                     :firstname => "u2", :auth_source_id => auth_sources(:one)).first_or_create
+    @u3 = User.where(:login => "u3", :mail => "u3@someware.com",
+                     :firstname => "u3", :auth_source_id => auth_sources(:one)).first_or_create
```

Comment:
Same issue here, it's searching for all of those attributes and not just the `login` field.